kube-core/src/metadata.rs

```diff
@@ -12,3 +12,23 @@ pub struct TypeMeta {
     /// The name of the API
     pub kind: String,
 }
+
+/// PartialObjectMetadata is a generic representation of any object with
+/// ObjectMeta. It allows clients to get access to a particular ObjectMeta
+/// schema without knowing the details of the version.
```

Comment:
```suggestion
/// A generic representation of any object with `ObjectMeta`.
///
/// It allows clients to get access to a particular `ObjectMeta`
/// schema without knowing the details of the version.
```
slight reformat for rustdoc